go/vt/sqlparser/comments.go

```diff
@@ -219,15 +221,15 @@ func ExtractCommentDirectives(comments Comments) CommentDirectives {
 			strVal := directive[sep+1:]
 			directive = directive[:sep]
 
-			boolVal, err := strconv.ParseBool(strVal)
+			intVal, err := strconv.Atoi(strVal)
```

Comment:
I changed the order here, otherwise you can't set integer values with 1. That gets parsed as boolean. 

This is ok to do as `isSet` already checks for both int and bool.